src/app/avatar/heygen/index.tsx

```diff
@@ -0,0 +1,174 @@
+'use client'
+import React, { useState, useEffect, useRef, useCallback } from 'react';
+import { handleICE, newSession, repeat, startSession, stopSession } from './helper';
+
+// Interface for session info
+interface SessionInfo {
+  session_id: string;
+  sdp: RTCSessionDescriptionInit;
+  ice_servers2: RTCIceServer[];
+}
+
+
+export function HeyGenPage({text}:{text: string}) {
+  // State variables
+  const [status, setStatus] = useState<string>('');
+  const [sessionInfo, setSessionInfo] = useState<SessionInfo | null>(null);
+  const [talkingToAI, setTalkingToAI] = useState(false);
+  const peerConnectionRef = useRef<RTCPeerConnection | null>(null);
+  const [mediaCanPlay, setMediaCanPlay] = useState<boolean>(false);
+
+  // References to media elements
+  const mediaElementRef = useRef<HTMLVideoElement>(null);
+  const canvasElementRef = useRef<HTMLCanvasElement>(null); 
+
+  // Event handlers (with type annotations)
+  const createNewSession = async () => {
+    console.log('Creating new session... please wait');
+
+    // Simplified: Replace these with your actual logic to get these values
+    const avatar = ''; // Example, replace with actual
```

Comment:
Did you mean to fill in some values here?

Reply:
I think we can leave them blank for now